tests/aequilibrae/paths/test_route_choice.py

```diff
@@ -161,43 +171,126 @@ def test_round_trip(self):
             .reset_index(drop=True)
         )
 
-        table = table.to_pandas().sort_values(by=["origin id", "destination id"]).reset_index(drop=True)
+        table = table.sort_values(by=["origin id", "destination id"]).reset_index(drop=True)
 
         pd.testing.assert_frame_equal(table, new_table)
 
     def test_cost_results(self):
         np.random.seed(0)
         rc = RouteChoiceSet(self.graph)
         nodes = [tuple(x) for x in np.random.choice(self.graph.centroids, size=(10, 2), replace=False)]
-        table = rc.batched(nodes, max_routes=20, max_depth=10, path_size_logit=True)
-        table = table.to_pandas()
+        rc.batched(nodes, max_routes=20, max_depth=10, path_size_logit=True)
+
+        table = rc.get_results().to_pandas()
 
         gb = table.groupby(by=["origin id", "destination id"])
         for od, df in gb:
             for route, cost in zip(df["route set"].values, df["cost"].values):
-                np.testing.assert_almost_equal(self.graph.cost[route].sum(), cost, err_msg=f"Cost differs for OD {od}")
+                np.testing.assert_almost_equal(
+                    self.graph.network.set_index("link_id").loc[route][self.graph.cost_field].sum(),
+                    cost,
+                    err_msg=f", cost differs for OD {od}",
+                )
 
-    def test_gamma_results(self):
+    def test_path_overlap_results(self):
         np.random.seed(0)
         rc = RouteChoiceSet(self.graph)
         nodes = [tuple(x) for x in np.random.choice(self.graph.centroids, size=(10, 2), replace=False)]
-        table = rc.batched(nodes, max_routes=20, max_depth=10, path_size_logit=True)
-        table = table.to_pandas()
+        rc.batched(nodes, max_routes=20, max_depth=10, path_size_logit=True)
+        table = rc.get_results().to_pandas()
 
         gb = table.groupby(by=["origin id", "destination id"])
         for od, df in gb:
-            self.assertTrue(all((df["gamma"] > 0) & (df["gamma"] <= 1)))
+            self.assertTrue(all((df["path overlap"] > 0) & (df["path overlap"] <= 1)))
 
     def test_prob_results(self):
         np.random.seed(0)
         rc = RouteChoiceSet(self.graph)
         nodes = [tuple(x) for x in np.random.choice(self.graph.centroids, size=(10, 2), replace=False)]
-        table = rc.batched(nodes, max_routes=20, max_depth=10, path_size_logit=True)
-        table = table.to_pandas()
+        rc.batched(nodes, max_routes=20, max_depth=10, path_size_logit=True)
+        table = rc.get_results().to_pandas()
 
         gb = table.groupby(by=["origin id", "destination id"])
         for od, df in gb:
-            self.assertAlmostEqual(1.0, sum(df["probability"].values), msg="Probability not close to 1.0")
+            self.assertAlmostEqual(1.0, sum(df["probability"].values), msg=", probability not close to 1.0")
+
+    def test_link_loading(self):
```

Comment:
Can we add a test where we know the answer, like for one od-pair and demand 1, where the link flows should should match the sum of probabilities across paths in each route set? We can use a small number of paths, max_depth=3 maybe.

Reply:
I've added two tests with known link loading results, one which has no demand between the selected OD pairs, and one with demand of 1 between single pair over a single link. The difference in cost is large enough that the probabilites come out binary so I've just asserted against 0.0 and 1.0, perhaps we should normalise the link costs so the probabilies are less extreme